src/cargo/sources/registry/remote.rs

```diff
@@ -175,12 +155,6 @@ impl<'cfg> RegistryData for RemoteRegistry<'cfg> {
         if self.needs_update {
             return Poll::Pending;
         }
-        // Check if the cache is valid.
-        let git_commit_hash = self.current_version();
-        if index_version.is_some() && index_version == git_commit_hash.as_deref() {
-            // This file was written by an old version of cargo, but it is still up-to-date.
-            return Poll::Ready(Ok(LoadResponse::CacheValid));
-        }
```

Comment:
Are we losing much by dropping this early return?

Reply:
See https://github.com/rust-lang/cargo/pull/11044 for more context.
Fundamentally, this will not be able to read files made by Cargo from before #10507. It will assume they are out of date and gen a new file. This does not feel like a big deal to me. But it can be added back, by reintroducing `self.head` and setting it to `resolve_to_commit()`, if you think that is better.

Reply:
Thanks for the explanation!